pendingBlock.go

```diff
@@ -165,13 +165,12 @@ func (b *pendingBlock) ExecuteNextTransaction(
 		return nil, err
 	}
 
-	if tp.Err == nil {
-		b.events = append(b.events, tp.Events...)
-		err := b.ledgerView.MergeView(childView)
-		if err != nil {
-			// fail fast if fatal error occurs
-			return nil, err
-		}
+	b.events = append(b.events, tp.Events...)
+
+	err = b.ledgerView.MergeView(childView)
+	if err != nil {
+		// fail fast if fatal error occurs
+		return nil, err
```

Comment:
@ramtinms Could you please have a look and check this matches the real behaviour in flow-go?

Reply:
The change is good IMO. A transaction error should not prevent merging the state or events.

But maybe @ramtinms can take a peek as well